### ISSUE TITLE
v:resource.record.fal does not respect workspaces at all

### ISSUE THREAD

**Issue by mkrappitz** [FluidTYPO3/vhs#896]
Given the following EXT:fluidcontent content element for example:

```html
{namespace flux=FluidTYPO3\Flux\ViewHelpers}
{namespace v=FluidTYPO3\Vhs\ViewHelpers}

<f:layout name="Content" />

<f:section name="Configuration">
	<flux:form id="image" label="Image" description="Image" options="{Fluidcontent: {sorting: 05}, group: 'FCE'}">
		<flux:field.inline.fal name="images" label="Image" minItems="1" maxItems="1" collapseAll="TRUE" />
	</flux:form>
</f:section>

<f:section name="Preview">
	...
</f:section>

<f:section name="Main">
	<f:for each="{v:content.resources.fal(field: 'images')}" as="image">
		<f:image treatIdAsReference="TRUE" src="{image.id}" alt="{image.alternative}" title="{image.title}" />
	</f:for>
</f:section>
```

it all works fine in the live workspace editing and creating such elements. Once in a draft workspace editing existing elements or creating new ones of this kind, v:content.resources.fal which internally triggers v:resource.record.fal fails to handle the sys_file_reference records correctly. It seems very obvious to me why that is the case once you look into the source of EXT:vhs/Classes/ViewHelpers/Resource/Record/FalViewHelper.php method getResources($record). There is no handling of workspaces in the sql query on line 57 at all, which results in the call of $this->getResource($row['uid']) on line 61 to fail and the $resources result becoming invalid.

Tested on TYPO3 CMS 6.2.14 and 7.4.0
EXT:vhs 2.3.3
EXT:fluidcontent 4.3.1

**Comment by mkrappitz** [comment]
It might also be the case that this error would not be existing if flux #898 would get fixed. But this is just speculation. Anyway I think these issues are connected.

**Comment by tiwden** [comment]
Not sure weather this is still helpfull, but we expirienced some problematic behaviour if the viewhelper is used in backend scope, for example in preview section. If there is a Live Record set to VersionState::DELETE_PLACEHOLDER the Record is still fetched and shown in the preview section.

to Prevent this, we changed 

`
foreach ($images as $file) {
  $resources[] = $this->getResource($file);
}`

to

`
foreach ($images as $file) {
  if($file->getProperty('t3ver_state') !== VersionState::DELETE_PLACEHOLDER){
    $resources[] = $this->getResource($file);
  };
}`

starting at line 124 in \vhs\Classes\ViewHelpers\Resource\Record\FalViewHelper.php

So obviously the workspace handling in the ResourceFactory is not working as expected.

Hope this could be helpfull to someone else who encounters this error
